coq/coq-system.el

```diff
@@ -39,7 +48,9 @@ This must be a single program name with no arguments; see `coq-prog-args'
 to manually adjust the arguments to the Coq process.
 See also `coq-prog-env' to adjust the environment."
   :type 'string
-  :group 'coq)
+  :group 'coq
+  :safe (lambda (a) (and (stringp a)
+                    (and (boundp 'proof-prog-name-ask) proof-prog-name-ask))))
```

Comment:
`and` can take more than two arguments, so no need to repeat the `and` here; also, there's a macro for this: `bound-and-true-p`

Reply:
>and can take more than two arguments, so no need to repeat the and here; also, there's a macro for this: bound-and-true-p

Sure.